### PR TITLE
estimate fee should use correct version

### PR DESCRIPTION
**Pull request by gaetbout**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Feature

## What is the current behaviour?

Estimate_fee accepts any version.

Resolves: #991 

## What is the new behaviour?

Estimate_fee should only accept the correct version according to: 
https://docs.starknet.io/documentation/architecture_and_concepts/Network_Architecture/Blocks/transactions/

## Does this introduce a breaking change?

Yes, all application should now use the correct version when estimating a transaction: 
`actual_tx_version_number + 0x100000000000000000000000000000000`
(query version should be set to true)

## Other information

Even though the CI is red all tests are passing locally.
Some tx version are/will be deprecated in the future. It should be updated accordingly.

### CODE REVIEW

**Comment by gaetbout** [comment]
Can't assign any reviewer, so just pinging @tdelabro

**Review by tdelabro — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -47,3 +47,5 @@
 - feat: add a `genesis_loader` for the node and mocking
 - feat: add `madara_tsukuyomi` as a submodule
 - branding: use new logo in the README
+- fix: estimate_fee should make sure all transaction have a version being
```
[the remark this answers is not in the document]
Comment (gaetbout) [reply CHANGELOG.md]: Moved

File: crates/pallets/starknet/src/tests/declare_tx.rs [hunk 2]
```diff
@@ -265,12 +265,13 @@ fn given_contract_declare_on_cairo_1_no_validate_account_then_it_works() {
...
             max_fee: Felt252Wrapper::from(u128::MAX),
             signature: bounded_vec!(),
-            is_query: false,
+            is_query: true,
         };
 
         let chain_id = Starknet::chain_id();
         let transaction_hash = calculate_declare_tx_hash(transaction.clone(), chain_id);
         transaction.signature = sign_message_hash(transaction_hash);
+        transaction.is_query = false;
```
[the remark this answers is not in the document]
Comment (gaetbout) [reply crates/pallets/starknet/src/tests/declare_tx.rs]: Indeed, my bad, fixed!

File: docs/faucet-setup.md [hunk 3]
```diff
@@ -121,7 +107,7 @@ async function transfer(to) {
     },
   ]);
   const maxFee = "0x11111111111";
-  const version = "0x1";
+  const version = "0x100000000000000000000000000000001";
```
[the remark this answers is not in the document]
Comment (gaetbout) [reply docs/faucet-setup.md]: Woopsie, reverted :D

**Review by tdelabro — APPROVED** [review]

**Pull request merged**